lib/multi.c

```diff
@@ -789,12 +805,22 @@ CURLMcode curl_multi_remove_handle(struct Curl_multi *multi,
   /* Remove the association between the connection and the handle */
   Curl_detach_connnection(data);
 
+  if(data->state.lastconnect) {
+    /* Mark any connect-only connection for closure */
+    Curl_conncache_foreach(data, data->state.conn_cache,
+                           data, &close_connect_only);
```

Comment:
If the check and action is done before `Curl_detach_connnection` is called, can't you then avoid having to iterate over all connections? Since this is done in all `remove_handle` calls there really can't be more than this particular connection to close, can it?

Reply:
There can't be multiple connections to close, but it's possible that the `lastconnect` connection was previously closed and freed, so we need to go through to make sure the connection still exists and that `lastconnect` points to valid memory.  If there's a sub-linear way to check whether `lastconnect` points to a connection in the cache we should use that instead, but I didn't see one.

Reply:
Fair enough. Thanks for looking. We can always come back and optimize this if we can come up with a better approach down the line...